### ISSUE TITLE
versioning a resource fails when author is no longer hs user

### ISSUE THREAD

**Issue by devincowan** [hydroshare/hydroshare#5275]
**Description of the bug**
Try to version this resource: https://beta.hydroshare.org/resource/2836494ee75e43a9bfb647b37260e461/
```
Traceback (most recent call last):
  File "/hydroshare/hs_core/tasks.py", line 803, in create_new_version_resource_task
    utils.copy_and_create_metadata(ori_res, new_res)
  File "/hydroshare/hs_core/hydroshare/utils.py", line 553, in copy_and_create_metadata
    dest_res.metadata.copy_all_elements_from(src_res.metadata, exclude_elements)
  File "/hydroshare/hs_core/models.py", line 4637, in copy_all_elements_from
    self.create_element(element_name, **element_args)
  File "/hydroshare/hs_core/models.py", line 4788, in create_element
    element = model_type.model_class().create(**kwargs)
  File "/hydroshare/hs_core/models.py", line 551, in create
    validate_hydroshare_user_id(hs_user_id)
  File "/hydroshare/hs_core/models.py", line 137, in validate_hydroshare_user_id
    raise ValidationError(err_message)
django.core.exceptions.ValidationError: ['1829 is not a valid id for hydroshare user']

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 451, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/celery/app/trace.py", line 734, in __protected_call__
    return self.run(*args, **kwargs)
  File "/hydroshare/hs_core/tasks.py", line 836, in create_new_version_resource_task
    raise utils.ResourceVersioningException(str(ex))
hs_core.hydroshare.utils.ResourceVersioningException: ['1829 is not a valid id for hydroshare user']
```

The resourcemetadata.xml contains:
```
    <dc:creator>
      <rdf:Description>
        <hsterms:organization>UT Austin Center for Water and the Environment</hsterms:organization>
        <hsterms:phone></hsterms:phone>
        <hsterms:hydroshare_user_id rdf:datatype="http://www.w3.org/2001/XMLSchema#integer">1829</hsterms:hydroshare_user_id>
        <hsterms:description rdf:datatype="http://www.w3.org/2001/XMLSchema#integer">1829</hsterms:description>
        <hsterms:creatorOrder rdf:datatype="http://www.w3.org/2001/XMLSchema#integer">2</hsterms:creatorOrder>
        <hsterms:homepage></hsterms:homepage>
        <hsterms:address></hsterms:address>
        <hsterms:name>Erika Boghici</hsterms:name>
        <hsterms:email>e.boghici@austin.utexas.edu</hsterms:email>
      </rdf:Description>
    </dc:creator>
```
The `hydroshare_user_id` is no longer valid. That causes the versioning to fail

**Expected behavior**
Versioning should succeed

**Additional information**
HS 2.11.1

**Comment by devincowan** [comment]
@martinseul suggested: "we'll need to prevent that a user who is an author on a published resource can be deleted"

**Comment by devincowan** [comment]
released as 2.11.2
